src/com/esotericsoftware/kryonet/TcpConnection.java

```diff
@@ -200,9 +204,21 @@ public int send (Connection connection, Object object) throws IOException {
 		SocketChannel socketChannel = this.socketChannel;
 		if (socketChannel == null) throw new SocketException("Connection is closed.");
 		synchronized (writeLock) {
+			int lengthLength = serialization.getLengthLength();
+
+			//wait while buffer has enough free space.
+			//thats because message length will be put only after the whole message will be serialized
+            while (writeBuffer.capacity() < writeBuffer.position() + lengthLength + objectBufferSize) {
+                try {
+                    Thread.sleep(10);
+                } catch (InterruptedException e) {
+                    break;
+                }
+                writeToSocket();
+	        }
```

Comment:
Can't just sleep the writing thread, this would have to be a feature, disabled by default. The right way to handle this is to not send data so fast or to use a larger buffer.

Comment should be "wait **until** buffer".